postgres/datadog_checks/postgres/util.py

```diff
@@ -212,9 +212,10 @@ def payload_pg_version(version):
 }
 
 q1 = (
-    'CASE WHEN pg_last_wal_receive_lsn() IS NULL OR '
-    'pg_last_wal_receive_lsn() = pg_last_wal_replay_lsn() THEN 0 ELSE GREATEST '
-    '(0, EXTRACT (EPOCH FROM now() - pg_last_xact_replay_timestamp())) END'
+    'CASE WHEN (SELECT count(*) FROM pg_stat_wal_receiver)>0 '
```

Comment:
I'm slightly confused with the intention described in the PR vs. this line of change `SELECT count(*) FROM pg_stat_wal_receiver)>0 AND ...`. Please correct me if I misunderstand 🙇 

As you described in `Additional Notes`, row counts > 0 in `pg_stat_wal_receiver` will indicate that native replication is ongoing. This check will effectively rule out archive-based replication. Somehow when I read the PR description, I got the feeling that the goal behind this PR is to include archive-based replication.

The logic is not wrong per se if change is to only calculate replication lag for transitional streaming replication.

Reply:
Row counts (10) > 0 in `pg_stat_wal_receiver` (native replication)
Row counts (0) > 0 in `pg_stat_wal_receiver` (archive-based replication)

In the previous version of the script, the query only assumed the usage of native replication, and the `pg_last_wal_receive_lsn()` function was always returning NULL in archive-based replication. This led to incorrect calculations, with the lag always being reported as 0.

To address this issue, a new line was introduced in the script to determine whether the replication method is native streaming or archive-based. The condition `(SELECT count(*) FROM pg_stat_wal_receiver) > 0` checks whether there is at least one active WAL receiver. If this condition is true (count greater than 0), it indicates native streaming replication, as only native replication sets up a WAL receiver. On the other hand, if the condition evaluates to false `(count is 0)`, it means archive-based replication.

By introducing this conditional check, the script is now able to correctly calculate the replication lag based on the active replication method. If the condition is false (archive-based), the script proceeds to calculate the lag. This approach ensures that the lag calculation is appropriate for both native streaming and archive-based replication scenarios.

This modification allows the script to adapt to the specific replication method being used, ensuring accurate lag calculations regardless of whether native streaming or archive-based replication is employed.

Reply:
Thanks for the detailed explanation 🙏